poetry/core/version/pep440/parser.py

```diff
@@ -61,7 +61,9 @@ def _get_local(cls, match: Optional[Match[AnyStr]]) -> Optional[LocalSegmentType
         )
 
     @classmethod
-    def parse(cls, value: str, version_class: Optional[Type["PEP440Version"]] = None):
+    def parse(
+        cls, value: str, version_class: Optional[Type["PEP440Version"]] = None
+    ) -> "PEP440Version":
```

Comment:
if you remove the `if TYPE_CHECKING` guard, this can be written as-

```suggestion
    def parse(
        cls, value: str, version_class: Optional[PEP440Version] = None
    ) -> PEP440Version:
```